src/mono/wasm/debugger/tests/debugger-test/debugger-evaluate-test.cs

```diff
@@ -833,9 +833,26 @@ public TestClass()
             }
         }
 
-        public static void Evaluate()
+        public struct TestStruct
```

Comment:
`debugger-get-properties-test.cs` has bunch of classes with inheritance, and mix of properties, and fields, and overrides etc, which you can also use for these tests.